src/tokens/TokenImport.tsx

```diff
@@ -63,13 +68,41 @@ export default function TokenImportScreen(_: Props) {
       if (supportedTokens[tokenId]) {
         setAddressState(AddressState.AlreadySupported)
       } else {
-        setAddressState(AddressState.Valid)
+        validateCheck(address)
       }
     } else {
       setAddressState(AddressState.Invalid)
     }
   }
 
+  const validateCheck = async (address: Address) => {
+    setAddressState(AddressState.FetchingContract)
+
+    const contract = getContract({
+      abi: erc20.abi,
+      address,
+      publicClient: publicClient[networkIdToNetwork[networkId]],
+    })
+
+    const state = await Promise.race<AddressState>([
```

Comment:
what do you think about splitting up the form validation and address fetch? it seems like they are different topics - one is a static check based only on user input, and one is asynchronous which could fail for reasons unrelated to the user.

i would propose that onBlur / on paste -> address validation happens, if there are no errors, then proceed to call the async function to fetch the details.

for async actions, we've generally used the `react-async-hook` lib which helps us to handle edge cases like the screen unmounting before all async actions have finished. this helps to avoid unresolved promises. i think we should use that here too